lib/bandit/http1/adapter.ex

```diff
@@ -73,8 +73,8 @@ defmodule Bandit.HTTP1.Adapter do
       {:ok, :http_eoh, rest} ->
         {:ok, headers, method, request_target, %{req | state: :headers_read, buffer: rest}}
 
-      {:ok, {:http_error, _reason}, _rest} ->
-        {:error, :invalid_request}
+      {:ok, {:http_error, reason}, _rest} ->
+        {:error, reason}
```

Comment:
Since the `reason` is (or can be) a list of integers, just logging the reason is not very helpful, and it is hard to filter/group similar log messages. Can the `{:http_error, reason}` tuple be returned as a whole here, or can the reason be prefixed with `http_error` in other ways otherwise?

Reply:
Yep. I wrote this before I'd seen the structure of the error (like I said, been trying to repro this for a while). This will most likely change based on the error you'd sent me privately.